kaiax/builder/impl/getter.go

```diff
@@ -100,3 +200,162 @@ func (b *BuilderModule) IsConflict(prevBundles []*builder.Bundle, newBundles []*
 
 	return false
 }
+
+// Filter removes elements at indices specified in `toRemove`.
+func Filter[T any](slice *[]T, toRemove map[int]bool) []T {
+	ret := make([]T, 0)
+	for i := 0; i < len(*slice); i++ {
+		if !toRemove[i] {
+			ret = append(ret, (*slice)[i])
+		}
+	}
+	return ret
+}
+
+func FindBundleIdx(bundles []*builder.Bundle, tx *types.Transaction) int {
+	for i, bundle := range bundles {
+		if bundle.Has(tx.Hash()) {
+			return i
+		}
+	}
+	return -1
+}
+
+func FindBundleIdxAsTxOrGen(bundles []*builder.Bundle, txOrGen interface{}) int {
+	for i, bundle := range bundles {
+		for _, txOrGenInBundle := range bundle.BundleTxs {
+			if EqualTxOrGen(txOrGenInBundle, txOrGen) {
+				return i
+			}
+		}
+	}
+	return -1
+}
+
+func EqualTxOrGen(txOrGenIX, txOrGenIY interface{}) bool {
+	var (
+		txOrGenXHash common.Hash
+		txOrGenYHash common.Hash
+	)
+
+	switch txOrGenX := txOrGenIX.(type) {
+	case *types.Transaction:
+		txOrGenXHash = txOrGenX.Hash()
+	case builder.TxGenerator:
+		txOrGenXHash = txOrGenX.Hash
+	}
+
+	switch txOrGenY := txOrGenIY.(type) {
+	case *types.Transaction:
+		txOrGenYHash = txOrGenY.Hash()
+	case builder.TxGenerator:
+		txOrGenYHash = txOrGenY.Hash
+	}
+
+	return bytes.Equal(txOrGenXHash.Bytes(), txOrGenYHash.Bytes())
+}
+
+func SetCorrectTargetTxHash(bundles []*builder.Bundle, txs []interface{}) []*builder.Bundle {
+	ret := make([]*builder.Bundle, 0)
+	for _, bundle := range bundles {
+		bundle.TargetTxHash = FindTargetTxHash(bundle, txs)
+		ret = append(ret, bundle)
+	}
+	return ret
+}
+
+func FindTargetTxHash(bundle *builder.Bundle, txs []interface{}) common.Hash {
```

Comment:
To look similar as `Find*` functions above:

```suggestion 
func FindTxIdx(txs []interface{}, tx *types.Transaction) int {
	...
}
```

Usage:
```
for _, bundle := range bundles {
	newTargetTx = txs[FindTxIdx(txs, bundle.BundleTxs[0]) - 1]
	bundle.TargetTxHash = newTargetTx.Hash()
	ret = append(ret, bundle)
}
```

Reply:
This usage may be possible in the future, but newTargetTx is an interface{} and does not have Hash().

Reply:
`newTargetTx` need to be type-asserted to `*types.Transaction`, because we know for sure that TargetTxHash can only be the type.

Reply:
Is the tx at index-1 in the beginning of the bundle guaranteed to be *types.Transaction?

Reply:
yes, because bundle can only be placed after tx.

Reply:
The generator may become a target due to the effects of pop etc.
e.g) 
[[A.tx1, A.tx2] [B.tx1, B.tx2, g] A.tx3 [C.tx1, C.tx2]]
if A.tx1 failed
[[B.tx1, B.tx2, g] [C.tx1, C.tx2]]